certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,209 @@
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+    bool otherAccountIsGuardian = currentContract.isGuardian(safeContract, otherAccount);
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;
 
+    currentContract.revokeGuardianWithThreshold(e, safeContract, prevGuardian, guardian, threshold);
+
+    assert safeContract.isModuleEnabled(currentContract);
+    assert e.msg.sender == safeContract;
+    assert !currentContract.isGuardian(safeContract, guardian);
+    assert guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian;
+    assert guardian != otherAccount => otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount > 0 => currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+    assert threshold <= guardianStorageContract.entries[safeContract].count;
+}
+
+// This integrity rule verifies that the guardian can always be revoked considering ideal conditions.
+rule guardianCanAlwaysBeRevoked(env e, address guardian, address prevGuardian, uint256 threshold) {
     requireSocialRecoveryModuleEnabled();
+    requireGuardiansLinkedListIntegrity(guardian);
+
+    // No value should be sent with the transaction.
+    require e.msg.value == 0;
+    // Guardian should not be zero address.
+    require guardian != 0;
+    // If new threshold is 0, then the guardian count should be 1.
+    require threshold == 0 => guardianStorageContract.entries[safeContract].count == 1;
+    // Else, if new threshold is greater than 0, then the guardian count should be greater than the threshold.
+    require threshold > 0 => guardianStorageContract.entries[safeContract].count > threshold;
+    // The address should be a guardian.
+    require currentContract.isGuardian(safeContract, guardian);
+
+    address nextGuardian = guardianStorageContract.entries[safeContract].guardians[guardian];
+    require guardianStorageContract.entries[safeContract].guardians[prevGuardian] == guardian;
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;    
+
+    // Safe Contract should be the sender of the transaction.
+    require e.msg.sender == safeContract;
+    currentContract.revokeGuardianWithThreshold@withrevert(e, safeContract, prevGuardian, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert !isReverted &&
+        guardianStorageContract.entries[safeContract].guardians[prevGuardian] == nextGuardian &&
+        !currentContract.isGuardian(safeContract, guardian) &&
+        currentGuardiansCount - 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies the possibilites in which the revocation of a new guardian can revert.
+rule revokeGuardianRevertPossibilities(env e, address otherAccount, address prevGuardian, uint256 threshold) {
```

Comment:
```suggestion
rule revokeGuardianRevertPossibilities(env e, address guardian, address prevGuardian, uint256 threshold) {
```

Reply:
Updated 👍🏾 my initial thought behind the `otherAccount` instead of `guardian` was that it could be some account that is not a guardian.